conf/migrations.py

```diff
@@ -35,3 +35,45 @@ def migration_231129_deploy_workflow(settings, data):
             logger.info(
                 f'Migrated {product_type}.DEPLOY_WORKFLOW to {product_type}.DEPLOY_WORKFLOWS'
             )
+
+
+def is_ipv4_url(text):
+    if isinstance(text, str) and 'ipv4' in text:
+        # Did not find the better way to filter only URLs so skipping it simple
+        # and open for reviewers suggestions
+        if 'redhat.com' in text:
+            return True
+    return False
+
+
+def ipv6_translator(settings_list, setting_major, settings):
+    dotted_settings = '.'.join(setting_major)
+    for _key, _val in settings_list.items():
+        if is_ipv4_url(_val):
+            settings.set(f'{dotted_settings}.{_key}', str(_val).replace('ipv4', 'ipv6'))
+            logger.debug(f'Setting translated to IPv6, Path: {dotted_settings}.{_key}')
+        elif type(_val) is list:
+            updated = False
+            new_list = _val
+            for i in range(len(new_list)):
+                if is_ipv4_url(new_list[i]):
+                    new_list[i] = new_list[i].replace('ipv4', 'ipv6')
+                    updated = True
+            if updated:
+                settings.set(f'{dotted_settings}.{_key}', new_list)
+                logger.debug(f'Setting translated to IPv6, Path: {dotted_settings}.{_key}')
+        elif type(_val) is dict:
+            new_setting_major = setting_major + [_key]
+            ipv6_translator(settings_list=_val, setting_major=new_setting_major, settings=settings)
+
+
+def migration_240426_ipv6_hostname_translation(settings, data):
```

Comment:
do I understand it correctly, that this turns everything into `ipv4` or `ipv6` based on a single setting?
aren't we effectively cutting ourselves off from eventual support of mixed stack in the future (e.g. ipv6 satellite test with some ipv4 resource)?

Reply:
I also believe that config migrations have been introduced to serve different purpose than this.

Another issue i am having with this approach is that you are hardcoding our internal infra design decision into the project (it was our infra decision to keep same hostnames with the changing `ipv` component in its name)